config/jobs/openenclave/jenkins/UbuntuContainerBuild.Jenkinsfile

```diff
@@ -0,0 +1,125 @@
+                        } finally {
+                            runner.cleanup()
+                        }
+                    }
+                }
+            }
+        }
+
+        // Run E2E Check if enabledTemporarily run always as e2e
+        stage('Install Prereqs (optional)') {
+            steps{
+                script{
+                    stage("Ubuntu ${params.LINUX_VERSION} Build - Install Prereqs") {
+                        def runner = load pwd() + "${SHARED_LIBRARY}"
+                        if("${params.E2E}" == "ON") {
+                            stage("${params.LINUX_VERSION} Setup") {
+                                try{
+                                    runner.cleanup()
+                                    runner.checkout("${params.PULL_NUMBER}")
+                                    runner.installOpenEnclavePrereqs()
+                                } catch (Exception e) {
+                                    // Do something with the exception 
+                                    error "Program failed, please read logs..."
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Go through Build stages
+        stage('Build') {
+            steps{
+                script{
+                    def runner = load pwd() + "${SHARED_LIBRARY}"
+
+                    // Build and test in Hardware mode, do not clean up as we will package
+                    stage("Ubuntu ${params.LINUX_VERSION} Build - ${params.BUILD_TYPE}") {
+                        try{
+                            runner.cleanup()
+                            runner.checkout("${params.PULL_NUMBER}")
+                            runner.cmakeBuildopenenclave("${params.BUILD_TYPE}","${params.COMPILER}","${EXTRA_CMAKE_ARGS}")
+                        } catch (Exception e) {
+                            // Do something with the exception 
+                            error "Program failed, please read logs..."
+                        }
+                    }
+
+                    // Build package and test installation work flows, clean up after
+                    stage("Ubuntu ${params.LINUX_VERSION} Package - ${params.BUILD_TYPE}") {
+                        try{
+                            runner.openenclavepackageInstall("${params.BUILD_TYPE}","${params.COMPILER}","${EXTRA_CMAKE_ARGS}")
+                        } catch (Exception e) {
+                            // Do something with the exception 
+                            error "Program failed, please read logs..."
+                        } finally {
+                            runner.cleanup()
+                        }
+                    }
+
+                    // Build in simulation mode 
+                    stage("Ubuntu ${params.LINUX_VERSION} Build - ${params.BUILD_TYPE} Simulation") {
+                        withEnv(["OE_SIMULATION=1"]) {
```

Comment:
You mentioned this is a dummy Jenkinsfile while you test, but I wanted to point out if you use OE_SIMULATION as a booleanParameter you may want to change this too, though this is particular instance is not important as it's just for consistency

Reply:
it's just a dummy file so will not be changing anything. I am not going to change the boolean parameter until later. If I do that I would need to change over 100 other files before I can get started on the feature I am working on.